src/main/java/unbanner/Section.java

```diff
@@ -246,26 +246,72 @@ public void setId(ObjectId id) {
     this.id = id;
   }
 
-  public boolean doesTimeConflictsRoom(){
+  @Override
+  public boolean equals(Object o){ //http://www.geeksforgeeks.org/overriding-equals-method-in-java/
+    if (o == this){
+      return true;
+    }
+    if (!(o instanceof Section)){
+      return false;
+    }
+    Section section = (Section) o;
+    return section.id.equals(this.id);
+  }
+
+  //Assumes that "sec" is on the same day and time
+  private boolean checkDayHours(Section sec){
+
+    //If the object (this) begins between any other section.
+    if (this.time.getFirst().isAfter(sec.getTime().getFirst()) && this.time.getFirst().isBefore(sec.time.getSecond()) ) {
+      return true;
+    }
+    //if this ends between any other section.
+    if (this.time.getSecond().isAfter(sec.getTime().getFirst()) && this.time.getSecond().isBefore(sec.time.getSecond())){
+      return true;
+    }
+    //if this is equal to any other section
+    if (this.time.getFirst().equals(sec.getTime().getFirst()) || this.time.getSecond().equals(sec.time.getSecond())){
+      return true;
+    }
+    //if this begins before another section and ends after that section
+    if (this.time.getSecond().isBefore(sec.getTime().getFirst()) && this.time.getSecond().isAfter(sec.time.getSecond())) {
+      return true;
+    }
+    return false;
+  }
+
+  public boolean doesTimeConflictsRoom(){  // some ".compare" and ".equals" methods would have be ideal
```

Comment:
Would you mind explaining the difference between this method and the already existing conflicts method?